src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -98,192 +89,270 @@ private InventoryUtils()
+    {
+        return hasItemInItemHandler(itemHandler, (ItemStack stack) -> Utils.checkIfPickaxeQualifies(requiredLevel, Utils.getMiningLevel(stack, Utils.PICKAXE)));
     }
 
     /**
-     * Checks if the inventory contains the following tool.
+     * Looks for a {@link ItemPickaxe} to mine a block of {@code requiredLevel},
+     * in the given {@link IItemHandler}. The {@link ItemPickaxe} tool level
+     * cannot exceed the given {@code maximalLevel}.
      *
-     * @param entity    the tileEntity chest or building.
-     * @param tool      the tool.
-     * @param toolLevel to check.
-     * @return true if found the tool.
+     * @param itemHandler   {@link IItemHandler} to check in.
+     * @param requiredLevel The minimal required {@link ItemPickaxe} level
+     * @param maximalLevel  The maximal tool level of the {@link ItemPickaxe}
+     * @return True if the {@link IItemHandler} contains a {@link ItemPickaxe}
+     * with the given minimal required level.
      */
-    public static boolean isToolInTileEntity(IInventory entity, final String tool, int toolLevel)
+    public static boolean isPickaxeInItemHandler(IItemHandler itemHandler, final int requiredLevel, final int maximalLevel)
     {
-        return InventoryFunctions.matchFirstInInventoryWithInventory(
-          entity,
-          stack -> Utils.isTool(stack, tool) && InventoryUtils.hasToolLevel(tool, stack, toolLevel),
-          InventoryFunctions::doNothing
-        );
+        return hasItemInItemHandler(itemHandler, (ItemStack stack) -> Utils.checkIfPickaxeQualifies(requiredLevel, Utils.getMiningLevel(stack, Utils.PICKAXE))
+                                                                        && InventoryUtils.hasToolLevel(stack, Utils.PICKAXE, maximalLevel));
     }
 
     /**
      * Verifies if there is one tool with an acceptable level
      * in a worker's inventory.
      *
-     * @param tool     the type of tool needed
-     * @param stack    the stack to test.
-     * @param hutLevel the worker's hut level
+     * @param toolName     the type of tool needed
+     * @param stack        the stack to test.
+     * @param minimalLevel the worker's hut level
      * @return true if tool is acceptable
      */
-    public static boolean hasToolLevel(final String tool, final ItemStack stack, final int hutLevel)
+    public static boolean hasToolLevel(@Nullable ItemStack stack, final String toolName, final int minimalLevel)
     {
-        if (stack == null || stack == ItemStack.EMPTY)
+        if (isItemStackEmpty(stack))
         {
             return false;
         }
 
-        final int level = Utils.getMiningLevel(stack, tool);
-        if (Utils.isTool(stack, tool) && verifyToolLevel(stack, level, hutLevel))
+        final int level = Utils.getMiningLevel(stack, toolName);
+        if (Utils.isTool(stack, toolName) && verifyToolLevel(stack, level, minimalLevel))
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException will be thrown when invoking method verifyToolLevel(). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)